src/test/java/bio/terra/workspace/integration/WorkspaceIntegrationTest.java

```diff
@@ -158,6 +163,49 @@ public void deleteWorkspaceWithInvalidToken(TestInfo testInfo) throws Exception
     Assertions.assertTrue(deleteWorkspaceResponse.isErrorObject());
   }
 
+  @Test
+  @Tag(TAG_NEEDS_CLEANUP)
+  public void createDataReference(TestInfo testInfo) throws Exception {
+    // This test relies on a persistent snapshot existing in Data Repo, currently in dev.
+    // This snapshot was created using our dev service account, which is a steward in dev Data Repo.
```

Comment:
Is there any blocker on creating the dataset/snapshot as part of this test code? I think tests should self-contain its resources whenever possible.
Also, I assume testing against non-dev environments of Data Repo is probably not a matter of `if`, but `when`?

Reply:
I don't think there's anything blocking us from creating the dataset as part of the test. That said, that's a lot of extra setup and teardown functionality that isn't really part of WM, and I'm a little concerned that there's a lot of room for test flakiness there. On the other hand, as implemented the test will start failing every time TDR clears their dev database (which should be rarely, but still). I'd need to think about this a bit

re: your second point, I'm not actually sure. At some point we'll definitely test against non-dev Data Repos, but will we be running these integration tests or something else?

Reply:
Coming back to this, I talked to Dan from TDR about this and he recommended using a persistent snapshot. Their dev environment is fairly stable and they make a reasonable effort to preserve the data, so failures due to deleted resources should be rare. Given that it removes a lot of setup/teardown complexity from this test (especially on functionality that isn't used elsewhere in WM), I'd prefer not to create a new snapshot each time we run the test.

Reply:
1. In that case, the current strategy works for me. I'd be interesting in learning what those extra setup/teardown functionality is for creating the dataset/snapshot, though.

2. Also, I'm wondering - after creating a data reference of a snapshot, are there any changes in the snapshot itself that we may need to undo/cleanup after the tests runs? Or does deleting the workspace which contains the data reference take care of any potential (_I don't know of any_) buildup in the snapshot?

3. To answer your question - Yes, I expect we'll be running these integration tests in other environments in the future.

Reply:
Also, would you mind adding the rationale above (i.e. `…Their dev environment is fairly stable and they make a reasonable effort to preserve the data…`) to the existing comments at the beginning of this `createDataReference` test?

Reply:
1. Creating the snapshot is three calls to TDR as the dev service account: creating a dataset (using [this schema](https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/test/resources/snapshot-test-dataset.json)), creating a snapshot on that dataset with the right settings, and then granting access to william.thunderlord (since we're running the TDR calls as the dev service account, rather than as william.thunderlord).
Cleaning up would be two delete calls, one for the snapshot and one for the dataset.
It's not super complex, but it would slow the test down and introduce a lot more surface area for errors that we shouldn't really be testing for

2. Snapshots in TDR are totally immutable(*), and WM's calls are all read-only.
(*) I think there's an exception for when patients request their data be removed and possibly another case, but from WM's perspective we should consider them immutable.

3. Nice, thanks! Updated comment